install_opi5.sh

```diff
@@ -4,55 +4,89 @@ if id "$1" >/dev/null 2>&1; then
     echo 'user found'
 else
     echo "creating pi user"
-    useradd pi -b /home
+    useradd pi -m -b /home -s /bin/bash
     usermod -a -G sudo pi
-    mkdir /home/pi
-    chown -R pi /home/pi
 fi
 echo "pi:raspberry" | chpasswd
 
-apt-get update
+apt-get update --quiet
+
+before=$(df --output=used / | tail -n1)
+# clean up stuff
+echo 'Purging snaps'
+# get rid of snaps
+rm -rf /var/lib/snapd/seed/snaps/*
+rm -f /var/lib/snapd/seed/seed.yaml
+apt-get purge --yes --quiet lxd-installer lxd-agent-loader
+apt-get purge --yes --quiet snapd
+
+# remove bluetooth daemon
+apt-get purge --yes --quiet bluez
+
+apt-get --yes --quiet autoremove
+
+after=$(df --output=used / | tail -n1)
+freed=$(( before - after ))
+
+echo "Freed up $freed KiB"
+
+# run Photonvision install script
 wget https://git.io/JJrEP -O install.sh
 chmod +x install.sh
 
-sed -i 's/# AllowedCPUs=4-7/AllowedCPUs=4-7/g' install.sh
+sed -i 's/# AllowedCPUs=4-7/AllowedCPUs=0-7/g' install.sh
 
-./install.sh -n -q
+./install.sh -m -q
 rm install.sh
 
+echo "Installing additional things"
 
-# Remove extra packages 
-echo "Purging extra things"
-# apt-get remove -y gdb gcc g++ linux-headers* libgcc*-dev
-# apt-get remove -y snapd
-apt-get autoremove -y
+apt-get install --yes --quiet network-manager net-tools libatomic1
 
+# let netplan create the config during cloud-init
+rm -f /etc/netplan/00-default-nm-renderer.yaml
 
-echo "Installing additional things"
-sudo apt-get update
-apt-get install -y network-manager net-tools libatomic1
+# set NetworkManager as the renderer in cloud-init
```

Comment:
I've got some reservations about the approach using cloud-init.

The upstream image exposes these on a separate device partition for editing on Windows. That's all fine, but it would be confusing for these files to be present yet not honored after first boot. IMO either keep cloud-init and always execute it (with a boot performance penalty) or just remove it entirely.

Reply:
My understanding of cloud-init is that it doesn't regenerate the network configuration on every boot, but I'll admit that it isn't entirely clear from the documentation. In addition, when photonvision changes the IP to static, NetworkManager adds it's own configuration file to `/etc/netplan` which will override any changes made in the cloud-init setup files.